### ISSUE TITLE
ConvLM

### ISSUE THREAD

**Issue by ahmadab** [facebookresearch/wav2letter#316]
I can see some recent changes related to enabling Decoding using ConvLM
Is there any documentation of how to build and use ConvLMs?

**Comment by tlikhomanenko** [comment]
Hi @ahmadab,

GCNN-14B architecture for ConvLM from the original [paper](https://arxiv.org/pdf/1612.08083.pdf) is defined [here](https://github.com/facebookresearch/wav2letter/blob/master/src/module/test/gcnn_14B_lm_arch_as.txt ) (you need to use adaptive softmax criterion to train this model). Besides this you can use [this arch](https://github.com/facebookresearch/wav2letter/blob/master/src/module/test/gcnn_14B_lm_arch_ce.txt ), GCNN-14B model which should be trained with cross-entropy criterion.

For decoder the ConvLM model should be similar to which you are using as acoustic model (in terms of creation and serializing).
Right now we are preparing to release several ConvLM models (Librispeech and WSJ) during upcoming weeks, so you can use them for decoding. Docs also is coming.

So far, during decoder you need to specify flag `lmtype=convlm`, `lm_vocab=path/to/file` in the format each row contain token/word which mapped into index of the row when lm was trained, `lm_memory=3000` (configure the total number of elements in an input matrix for lm forward pass).

Feel free to reopen if you need more info/details.

**Comment by ahmadab** [comment]
Thanks @tlikhomanenko

**Comment by hiroaki-ogawa** [comment]
Hi @tlikhomanenko,

I trained character based convlm of WSJ using fairseq. So I  have conlm in format of fairseq.
(lets say, checkpoints/checkpoint_best.pt)
I gave this lm to the decoder 
```
--lexicon ""
--lm checkpoints/checkpoint_best.pt
--lm_vocab path/to/dict.txt
--lmtype convlm
--lm_memory 5000
```
and got follwoing error
```
terminate called after throwing an instance of 'std::invalid_argument'
  what():  Unknown token in dictionary: <unk>
*** Aborted at 1565701299 (unix time) try "date -d @1565701299" if you are using GNU date ***
PC: @     0x7fc61031b428 gsignal
*** SIGABRT (@0xd5000002ab) received by PID 683 (TID 0x7fc65a18e600) from PID 683; stack trace: ***
    @     0x7fc617b9b390 (unknown)
    @     0x7fc61031b428 gsignal
    @     0x7fc61031d02a abort
    @     0x7fc610c5e84d __gnu_cxx::__verbose_terminate_handler()
    @     0x7fc610c5c6b6 (unknown)
    @     0x7fc610c5c701 std::terminate()
    @     0x7fc610c5c919 __cxa_throw
    @           0x50c22c w2l::Dictionary::getIndex()
    @           0x56397b w2l::ConvLM::ConvLM()
    @           0x487364 std::__shared_ptr<>::__shared_ptr<>()
    @           0x41a8b8 main
    @     0x7fc610306830 __libc_start_main
    @           0x4762d9 _start
    @                0x0 (unknown)
```
do I have to convert the conlm for wav2letters?
